lib/urbanopt/geojson/schema/building_properties.json

```diff
@@ -376,6 +476,36 @@
         "DEER 2070",
         "DEER 2075"
       ]
+    },
+    "foundationType": {
+      "description": "The foundation type of the building. Required for residential buildings.",
+      "type": "string",
+      "enum": [
+        "slab",
+        "crawlspace - vented",
+        "crawlspace - unvented",
+        "basement - unconditioned",
+        "basement - conditioned",
+        "ambient"
+      ]
+    },
+    "atticType": {
+      "description": "The attic type of the building. Required for residential buildings.",
```

Comment:
Attic type won't apply if there's a flat roof. And choice between gable and hip isn't really so important (other than geometry rendering). I suggest combining attic/roof choices into a single list:
- "attic - vented"
- "attic - unvented"
- "attic - conditioned"
- "flat roof"

And then just assume home has a gable roof if there's an attic.